examples/versions/index.js

```diff
@@ -0,0 +1,282 @@
+import { Value, Operation } from 'slate'
+import { Editor } from 'slate-react'
+
+import React from 'react'
+import styled from 'react-emotion'
+
+import { Stack } from 'immutable'
+
+import { Button, Icon, Toolbar } from '../components'
+
+const initialVersionState = [
+  {
+    name: 'version 1',
+    isRoot: true,
+    value: {
+      document: {
+        nodes: [
+          {
+            object: 'block',
+            type: 'paragraph',
+            nodes: [
+              {
+                object: 'text',
+                leaves: [
+                  {
+                    text: 'This example shows versions.',
+                  },
+                ],
+              },
+            ],
+          },
+        ],
+      },
+    },
+  },
+  {
+    name: 'version 2',
+    changes: [
+      [
+        Operation.create({
+          object: 'operation',
+          path: [0, 0],
+          position: 28,
+          properties: {
+            data: {},
+            type: undefined,
+          },
+          target: null,
+          type: 'split_node',
+        }),
+        Operation.create({
+          object: 'operation',
+          path: [0],
+          position: 1,
+          properties: {
+            data: {},
+            type: 'paragraph',
+          },
+          target: 28,
+          type: 'split_node',
+        }),
+      ],
+      [
+        Operation.create({
+          marks: [],
+          object: 'operation',
+          offset: 0,
+          path: [1, 0],
+          text: 'Try adding a new version by clicking the + icon.',
+          type: 'insert_text',
+        }),
+      ],
+    ],
+  },
+]
+
+export const VersionList = styled('ul')``
+
+export const VersionListItem = styled('li')`
+  cursor: pointer;
+  color: ${props => (props.active ? 'red' : 'black')};
+`
+
+export const Version = ({ active, onClick, name }) => (
+  <VersionListItem active={active} onClick={onClick}>
+    {name}
+  </VersionListItem>
+)
+
+/**
+ * The versions example.
+ *
+ * @type {Component}
+ */
+
+class Versions extends React.Component {
+  /**
+   * Deserialize the initial editor value.
+   *
+   * @type {Object}
+   */
+
+  state = {
+    value: Value.fromJSON(initialVersionState[0].value),
+    versions: initialVersionState,
+    activeVersionIndex: 0,
+  }
+
+  componentDidMount() {
+    this.setVersion(initialVersionState.length - 1)
+  }
+
+  /**
+   * Resets the history stack
+   *
+   */
+
+  resetHistory = () => {
+    let { value } = this.state
+    const change = value.change()
+
+    const history = value.history
+      .set('undos', new Stack())
+      .set('redos', new Stack())
+    value = value.set('history', history)
+    change.value = value
+
+    this.onChange(change)
+  }
+
+  /**
+   * Sets a version as the active version
+   *
+   * @param {Number} index
+   */
+
+  setVersion = index => {
```

Comment:
A small suggestion.  Would you like to use `this.setState( state => )` to avoid calling `this.state`?  Or use 
```
this.setState(({ value, versions, activeVersionIndex }) => {...})
```